utils/docs/build.js

```diff
@@ -0,0 +1,178 @@
+var fs = require( "fs" );
+
+function JSONsort( object ) {
+
+	const isObject = ( objectToTest ) => ( '[object Object]' === Object.prototype.toString.call( objectToTest ) );
+
+	if ( Array.isArray( object ) ) {
+
+		return object.sort().map( item => isObject( item ) ? JSON.sort( item ) : item );
+
+	} else if ( isObject( object ) ) {
+
+		return Object
+			.keys( object )
+			.sort()
+			.reduce( ( obj, key ) => {
+
+				if ( isObject( object[ key ] ) ) {
+
+					obj[ key ] = JSON.sort( object[ key ] );
+
+				} else if ( Array.isArray( object[ key ] ) ) {
+
+					obj[ key ] = object[ key ].map( item => isObject( item ) ? JSON.sort( item ) : item );
+
+				} else {
+
+					obj[ key ] = object[ key ];
+
+				}
+
+				return obj;
+
+			}, {} );
+
+	}
+
+	return object;
+
+}
+
+function getParent( HTMLFile ) {
+
+
+	var ParentRegex = /\[page:([\w\.]+) ?([\w\.\s]+)?\]\s+&rarr;/gi;
+	do {
+
+		match = ParentRegex.exec( HTMLFile );
+		if ( match ) {
+
+			return match[ 1 ];
+
+		}
+
+	} while ( match );
+
+	return undefined;
+
+}
+
+function getMembers( HTMLFile ) {
+
+	var memberregex = /\[(?:member|property|method):([\w]+) ?([\w\.\s]+)?\]/gi;
+	var matches = [];
+	do {
+
+		match = memberregex.exec( HTMLFile );
+		if ( match && match[ 2 ] ) {
+
+			matches.push( match[ 2 ] );
+
+		} else if ( match && match[ 1 ] ) {
+
+			matches.push( match[ 1 ] );
+
+		}
+
+	} while ( match );
+	return matches;
+
+}
+
+
+function extractData( obj ) {
+
+	var allPages = {};
+
+	for ( var category in obj ) {
+
+		var pages = obj[ category ];
+		for ( var page in pages ) {
+
+			var filterItems = pages[ page ];
+			if ( typeof filterItems === "string" ) {
+
+				filterItems = { "#URL": filterItems };
+
+			} else {
+
+				filterItems = { "#URL": filterItems[ "#URL" ] };
+
+			}
+			var url = filterItems[ "#URL" ];
+			var HTMLFile = fs.readFileSync( docsFolder + url + ".html", "utf8" );
+
+			var parent = getParent( HTMLFile );
+			if ( parent ) {
+
+				filterItems[ "#PARENT" ] = parent;
+
+			}
+
+
+			var members = getMembers( HTMLFile );
+			if ( members && members.length > 0 ) {
+
+				members.forEach( function ( member ) {
+
+					filterItems[ member ] = page + "." + member;
+
+				} );
+
+			}
+
+			pages[ page ] = JSONsort( filterItems );
+
+			allPages[ page ] = filterItems;
+
+		}
+
+	}
+	/*
+    for (var pageToCheck in allPages) {
```

Comment:
Does this commented code need to stay?

Reply:
This code actually adds all methods from basetype to the list.js file too.

Reply:
Well, it currently does nothing. :smile: 

If this really is a needed feature, how about adding a command line switch to the script?
For example: `$ node utils/docs/build.js --include-base-type-methods`

Reply:
I've added the command line switch.